Flaky unit test DisposeMethodsShouldCallBaseClassDisposeTests.ErrorCase_NoDiagnostic

This test is occasionally failing in our signed/CI builds. Seems like compiler is sometimes reporting an extra CS error. We may want to consider updating/removing this test from this repo to workaround this (while also trying to identify a repro case for compiler issue).

**Failing test**: https://github.com/dotnet/roslyn-analyzers/blob/a23fde8d03fcec0c8e5beb6b1f3882048ab8435b/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposeMethodsShouldCallBaseClassDisposeTests.cs#L1083-L1117

**Test failure log:**
```
Microsoft.CodeAnalysis.Testing.Verifiers.EqualWithMessage
Exception : Context: Diagnostics of test state
Mismatch between number of diagnostics returned, expected "7" actual "8"

Diagnostics:
// /0/Test0.cs(2,29): error CS0246: El nombre del tipo o del espacio de nombres 'IComparable<>' no se encontró (¿falta una directiva using o una referencia de ensamblado?)
DiagnosticResult.CompilerError("CS0246").WithSpan(2, 29, 2, 43).WithArguments("IComparable<>"),
// /0/Test0.cs(3,16): error CS0246: El nombre del tipo o del espacio de nombres 'IComparable<>' no se encontró (¿falta una directiva using o una referencia de ensamblado?)
DiagnosticResult.CompilerError("CS0246").WithSpan(3, 16, 3, 30).WithArguments("IComparable<>"),
// /0/Test0.cs(3,16): error CS0246: El nombre del tipo o del espacio de nombres 'IComparable<>' no se encontró (¿falta una directiva using o una referencia de ensamblado?)
DiagnosticResult.CompilerError("CS0246").WithSpan(3, 16, 3, 30).WithArguments("IComparable<>"),
// /0/Test0.cs(10,22): error CS1061: "T" no contiene una definición para "CompareTo" ni un método de extensión accesible "CompareTo" que acepte un primer argumento del tipo "T" (¿falta alguna directiva using o una referencia de ensamblado?)
DiagnosticResult.CompilerError("CS1061").WithSpan(10, 22, 10, 31).WithArguments("T", "CompareTo"),
// /0/Test0.cs(13,26): error CS0115: 'BaseClass<T>.Equals(object)': no se encontró ningún miembro adecuado para invalidar
DiagnosticResult.CompilerError("CS0115").WithSpan(13, 26, 13, 32).WithArguments("BaseClass<T>.Equals(object)"),
// /0/Test0.cs(23,25): error CS0115: 'BaseClass<T>.GetHashCode()': no se encontró ningún miembro adecuado para invalidar
DiagnosticResult.CompilerError("CS0115").WithSpan(23, 25, 23, 36).WithArguments("BaseClass<T>.GetHashCode()"),
// /0/Test0.cs(26,14): error CS0314: El tipo 'T' no se puede usar como parámetro de tipo 'T' en el tipo o método genérico 'BaseClass<T>'. No hay conversión boxing ni conversión de parámetro de tipo de 'T' a 'IComparable<T>'.
DiagnosticResult.CompilerError("CS0314").WithSpan(26, 14, 26, 26).WithArguments("BaseClass<T>", "IComparable<T>", "T", "T"),
// /0/Test0.cs(27,15): error CS0246: El nombre del tipo o del espacio de nombres 'IComparable<>' no se encontró (¿falta una directiva using o una referencia de ensamblado?)
DiagnosticResult.CompilerError("CS0246").WithSpan(27, 15, 27, 29).WithArguments("IComparable<>"),

Assert.Equal() Failure\r\nExpected: 7\r\nActual:   8
```

```
    Microsoft.NetCore.Analyzers.Runtime.UnitTests.DisposeMethodsShouldCallBaseClassDisposeTests.ErrorCase_NoDiagnostic [FAIL]
      Microsoft.CodeAnalysis.Testing.Verifiers.EqualWithMessageException : Context: Diagnostics of test state
      Mismatch between number of diagnostics returned, expected "7" actual "8"
      
      Diagnostics:
      // /0/Test0.cs(2,29): error CS0246: The type or namespace name 'IComparable<>' could not be found (are you missing a using directive or an assembly reference?)
      DiagnosticResult.CompilerError("CS0246").WithSpan(2, 29, 2, 43).WithArguments("IComparable<>"),
      // /0/Test0.cs(3,16): error CS0246: The type or namespace name 'IComparable<>' could not be found (are you missing a using directive or an assembly reference?)
      DiagnosticResult.CompilerError("CS0246").WithSpan(3, 16, 3, 30).WithArguments("IComparable<>"),
      // /0/Test0.cs(3,16): error CS0246: The type or namespace name 'IComparable<>' could not be found (are you missing a using directive or an assembly reference?)
      DiagnosticResult.CompilerError("CS0246").WithSpan(3, 16, 3, 30).WithArguments("IComparable<>"),
      // /0/Test0.cs(10,22): error CS1061: 'T' does not contain a definition for 'CompareTo' and no accessible extension method 'CompareTo' accepting a first argument of type 'T' could be found (are you missing a using directive or an assembly reference?)
      DiagnosticResult.CompilerError("CS1061").WithSpan(10, 22, 10, 31).WithArguments("T", "CompareTo"),
      // /0/Test0.cs(13,26): error CS0115: 'BaseClass<T>.Equals(object)': no suitable method found to override
      DiagnosticResult.CompilerError("CS0115").WithSpan(13, 26, 13, 32).WithArguments("BaseClass<T>.Equals(object)"),
      // /0/Test0.cs(23,25): error CS0115: 'BaseClass<T>.GetHashCode()': no suitable method found to override
      DiagnosticResult.CompilerError("CS0115").WithSpan(23, 25, 23, 36).WithArguments("BaseClass<T>.GetHashCode()"),
      // /0/Test0.cs(26,14): error CS0314: The type 'T' cannot be used as type parameter 'T' in the generic type or method 'BaseClass<T>'. There is no boxing conversion or type parameter conversion from 'T' to 'IComparable<T>'.
      DiagnosticResult.CompilerError("CS0314").WithSpan(26, 14, 26, 26).WithArguments("BaseClass<T>", "IComparable<T>", "T", "T"),
      // /0/Test0.cs(27,15): error CS0246: The type or namespace name 'IComparable<>' could not be found (are you missing a using directive or an assembly reference?)
      DiagnosticResult.CompilerError("CS0246").WithSpan(27, 15, 27, 29).WithArguments("IComparable<>"),
      
      
      Assert.Equal() Failure
      Expected: 7
      Actual:   8
```

--- a/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposeMethodsShouldCallBaseClassDisposeTests.cs
+++ b/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposeMethodsShouldCallBaseClassDisposeTests.cs
@@ -1083,41 +1083,5 @@ End Class",
             // Test0.vb(14,26): warning CA2215: Ensure that method 'Sub B.Dispose()' calls 'MyBase.Dispose()' in all possible control flow paths.
             GetBasicResultAt(14, 26, "Sub B.Dispose()", "MyBase.Dispose()"));
         }
-
-        [Fact, WorkItem(1671, "https://github.com/dotnet/roslyn-analyzers/issues/1671")]
-        public async Task ErrorCase_NoDiagnostic()
-        {
-            // Missing "using System;" causes "Equals" method be marked as IsOverride but with null OverriddenMethod.
-            await VerifyCS.VerifyAnalyzerAsync(@"
-public class BaseClass<T> : {|CS0246:IComparable<T>|}
-     where T : {|CS0246:IComparable<T>|}
-{
-    public T Value { get; set; }
-
-
-    public int CompareTo(T other)
-    {
-        return Value.{|CS1061:CompareTo|}(other);
-    }
-
-    public override bool {|CS0115:Equals|}(object obj)
-    {
-        if (obj is BaseClass<T> other)
-        {
-            return Value.Equals(other.Value);
-        }
-
-        return false;
-    }
-
-    public override int {|CS0115:GetHashCode|}() => Value?.GetHashCode() ?? 0;
-}
-
-public class {|CS0314:DerivedClass|}<T> : BaseClass<T>
-    where T : {|CS0246:IComparable<T>|}
-{
-}
-");
-        }
     }
 }

